src/main/java/seedu/address/model/event/StartDateWithinTimeFramePredicate.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.model.event;
+
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Event}'s {@code startDateTime} is within a given time frame.
+ */
+public class StartDateWithinTimeFramePredicate implements Predicate<Event> {
+    private final LocalDate startDate;
+    private final LocalDate endDate;
+
+    /**
+     * Constructor for StartDateWithinTimeFramePredicate
+     */
+    public StartDateWithinTimeFramePredicate(LocalDate startDate, LocalDate endDate) {
+        this.startDate = startDate;
+        this.endDate = endDate;
+    }
+
+    /**
+     * Returns true when event starts after startDate and before or on endDate.
+     */
+    @Override
+    public boolean test(Event event) {
+        LocalDate eventStartDate = event.getStartDateTime().dateTime.toLocalDate();
```

Comment:
Might not be a good idea to access the internal structure of `startDateTime` using `.dateTime` here. But I think we should KIV and sort this out later since #100 and #106 has some changes for the `DateTime` class.